src/dynimage.rs

```diff
@@ -760,6 +760,59 @@ fn open_impl(path: &Path) -> ImageResult<DynamicImage> {
     load(fin, format)
 }
 
+/// Read the dimensions of the image located at the specified path.
+/// This is faster than fully loading the image and then getting its dimensions.
+pub fn image_dimensions<P>(path: P) -> ImageResult<(u32, u32)>
+where
+    P: AsRef<Path>
+{
+    // thin wrapper function to strip generics before calling open_impl
+    image_dimensions_impl(path.as_ref())
+}
+
+fn image_dimensions_impl(path: &Path) -> ImageResult<(u32, u32)> {
+    let fin = File::open(path)?;
+    let fin = BufReader::new(fin);
+
+    let ext = path
+        .extension()
+        .and_then(|s| s.to_str())
+        .map_or("".to_string(), |s| s.to_ascii_lowercase());
+
+    match &ext[..] {
+        #[cfg(feature = "jpeg")]
+        "jpg" | "jpeg" => Ok(jpeg::JPEGDecoder::new(fin)?.dimensions()),
+        #[cfg(feature = "png_codec")]
+        "png" => Ok(png::PNGDecoder::new(fin)?.dimensions()),
+        #[cfg(feature = "gif_codec")]
+        "gif" => Ok(gif::Decoder::new(fin)?.dimensions()),
+        #[cfg(feature = "webp")]
+        "webp" => Ok(webp::WebpDecoder::new(fin)?.dimensions()),
+        #[cfg(feature = "tiff")]
+        "tif" | "tiff" => Ok(tiff::TIFFDecoder::new(fin)?.dimensions()),
+        #[cfg(feature = "tga")]
+        "tga" => Ok(tga::TGADecoder::new(fin)?.dimensions()),
+        #[cfg(feature = "bmp")]
+        "bmp" => Ok(bmp::BMPDecoder::new(fin)?.dimensions()),
+        #[cfg(feature = "ico")]
+        "ico" => Ok(ico::ICODecoder::new(fin)?.dimensions()),
+        #[cfg(feature = "hdr")]
+        "hdr" => Ok(hdr::HDRAdapter::new(fin)?.dimensions()),
+        #[cfg(feature = "pnm")]
+        "pbm" | "pam" | "ppm" | "pgm" => {
+            Ok(pnm::PNMDecoder::new(fin)?.dimensions())
+        }
+        format => Err(image::ImageError::UnsupportedError(format!(
+            "Image format image/{:?} is not supported.",
+            format
+        ))),
+    }
+     // TODO possible overflow here, though not probable since it would
+     // require an extremely large image.
+    .map(|(w, h)| (w as u32, h as u32))
```

Comment:
While not probable, the correct way to handle them is to return an error of either `ImageError::DimensionError` or `ImageError::UnsupportedError` with explanation in these cases. Otherwise, an allocation may be larger or smaller than expected, or a malicious image could appear small but then consume a large amount of memory when actually loaded.

Reply:
Thanks for the review, that makes the function clearer to read and the overflowing behaviour is now more explicit.